compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -444,12 +447,15 @@ class TreeUnpickler(reader: TastyReader,
               readTypeRef() match {
                 case binder: LambdaType => binder.paramRefs(readNat())
               }
+            case _ if isBestEffortTasty =>
+              goto(end)
+              new PreviousErrorType
           }
         assert(currentAddr == end, s"$start $currentAddr $end ${astTagToString(tag)}")
         result
       }
 
-      def readSimpleType(): Type = (tag: @switch) match {
+      def readSimpleType(): Type = tag match {
```

Comment:
This should definitely not be removed. We want to ensure that this is compiled as a `switch_table` on the JVM.